geode-core/src/main/java/org/apache/geode/internal/cache/TXStateProxyImpl.java

```diff
@@ -678,7 +702,21 @@ public void setTarget(DistributedMember target) {
     if (currRegion.isUsedForPartitionedRegionBucket()) {
       return currRegion.getRegionKeysForIteration();
     } else {
-      return getRealDeal(null, currRegion).getRegionKeysForIteration(currRegion);
+      boolean resetTxState = isTransactionInternalSuspendNeeded(currRegion);
+      TXStateProxy txp = null;
+      if (resetTxState) {
```

Comment:
If !resetTxState, then the getRealDeal(...) action will not be called and nothing will be done in the finally block: consider refactoring that logic path out of the current logic for resetTxState being true in order to reduce complexity.

Reply:
This is currently the pattern how product does to internally suspend (pause) the transaction. If need to refactor, it will be another JIRA ticket.